cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -239,4 +240,20 @@ private PullAppResponse<?> pullAndValidateApplication(ApplicationReference appRe
     }
     return pullResponse;
   }
+
+  /**
+   * The method to list all applications found in linked repository.
+   *
+   * @return {@link RepositoryAppsResponse}
+   * @throws RepositoryNotFoundException   if the repository config is not found
+   * @throws AuthenticationConfigException if git auth config is not found
+   * @throws SourceControlException if {@link SourceControlOperationRunner} fails to list applications
+   */
+  public RepositoryAppsResponse listApps(NamespaceId namespace) throws NotFoundException,
```

Comment:
If we know it's `RepositoryNotFoundException`, let's specify it: `NotFoundException` --> `RepositoryNotFoundException`

Reply:
Makes sense. Will do